src/Analyzers/Core/Analyzers/UseCollectionInitializer/UpdateExpressionState.cs

```diff
@@ -369,7 +369,10 @@ public bool ValuePatternMatches(TExpressionSyntax expression)
 
         Match<TStatementSyntax>? TryAnalyzeForeachStatement(TStatementSyntax foreachStatement)
         {
-            syntaxHelper.GetPartsOfForeachStatement(foreachStatement, out var identifier, out _, out var foreachStatements);
+            syntaxHelper.GetPartsOfForeachStatement(foreachStatement, out var awaitKeyword, out var identifier, out _, out var foreachStatements);
+            if (awaitKeyword != default)
```

Comment:
`IsKind` is a faster check here

```suggestion
            if (awaitKeyword.IsKind(SyntaxKind.None))
```

Reply:
i personally prefer the 'default' check :)  it's very sensible to me, and akin to checking `null` for the reference type case :)